### DISCUSSION TITLE
The ei_data.payment_method is required when e_invoice is true

### DISCUSSION THREAD

**Discussion by lucabartoli** [fattureincloud/api#57] [Q&A] [answered]
Da stamattina la creazione di una fattura obbliga all'inserimento del payment_method anche nella fatturazione elettronica.
Il mio intento sarebbe quello di preparare una fattura per la revisione e invio e tutto ha funzionato correttamente fino a ieri sera, senza passare alcun ei_data.

Come mai questo cambiamento?
Essendo anche un cambiamento rilevante, esiste un posto dove verificare i changelog che corrompono funzionalità di integrazioni già sviluppate?

Grazie mille

**Comment by emajo** [comment, marked as answer]
Ciao @lucabartoli, si, abbiamo aggiunto questa validazione in quanto una fattura elettronica senza payment method non passerebbe la validazione dell'xml, questo comportamento previene errori nelle chiamate api verifyXml e sendEInvoice, probabilmente lei faceva revisione e invio dalla nostra interfaccia web che va a aggiustare automaticamente l'xml.
Stiamo pensando di creare un changelog su questa repository per tenere traccia dei cambiamenti alle nostre api.

Grazie per il suggerimento,
Emanuele

**Reply by lucabartoli** [reply]
Si effettivamente facevo esattamente in questo modo: preparazione e poi convalida e invio dall'interfaccia.
Mi sono aggiunto il campo nella request così ora passa.
Grazie mille per il supporto

**Reply by valmoz** [reply]
Se a qualcuno dovesse servire, qui spiega il campo da settare:
https://github.com/fattureincloud/api/discussions/25#discussioncomment-3269392

**Reply by lelebus** [reply]
Queste comunque sono breaking changes, non dovrebbero essere rilasciate cosi con leggerezza. Abbiamo decine di applicazioni da aggiornare e clienti incazzati perche da un momento all'altro si sono bloccati i servizi che abbiamo sviluppato e integrato con fattureincloud.

Secondo gli standard del versioning, breaking changes richiederebbero il rilascio di una nuova major version, cioe', si sarebbe dovuti passare alla V3.

Mi auguro veramente che non succeda mai piu un fatto cosi' spiacevole.
